src/shared_modules/utils/globHelper.h

```diff
@@ -41,7 +41,10 @@ namespace Utils
                     if (patternPos + 1 < pattern.size() && pattern.at(patternPos + 1) == entryName.at(i))
                     {
                         ++patternPos;
-                        --i;
+                        if (i > 0)
+                        {
+                            --i;
+                        }
```

Comment:
Good catch. Perhaps the following check could be added:

```c++
                    if (i < 0)
                    {
                        i = 0;
                    }
```

 It seems to work, I added some tests:
 
 https://godbolt.org/z/7caTcWcoz

Reply:
[This proposal](https://github.com/wazuh/wazuh/pull/25066/files#r1707306564) has no effect because i is an unsigned int

Maybe the method itself got into consideration this underflow

Reply:
Thanks for the review